Comp2023/src/main/java/frc/robot/subsystems/Swerve.java

```diff
@@ -209,142 +217,55 @@ private void updateSmartDashboard( )
   //
   // Limelight driving mode
   //
-  public void driveWithLimelightInit(boolean m_endAtTarget)
-  {
-    // get pid values from dashboard
-    m_turnConstant = SmartDashboard.getNumber("DTL_turnConstant", m_turnConstant);
-    m_turnPidKp = SmartDashboard.getNumber("DTL_turnPidKp", m_turnPidKp);
-    m_turnPidKi = SmartDashboard.getNumber("DTL_turnPidKi", m_turnPidKi);
-    m_turnPidKd = SmartDashboard.getNumber("DTL_turnPidKd", m_turnPidKd);
-    m_turnMax = SmartDashboard.getNumber("DTL_turnMax", m_turnMax);
-
-    m_throttlePidKp = SmartDashboard.getNumber("DTL_throttlePidKp", m_throttlePidKp);
-    m_throttlePidKi = SmartDashboard.getNumber("DTL_throttlePidKi", m_throttlePidKi);
-    m_throttlePidKd = SmartDashboard.getNumber("DTL_throttlePidKd", m_throttlePidKd);
-    m_throttleMax = SmartDashboard.getNumber("DTL_throttleMax", m_throttleMax);
-    m_throttleShape = SmartDashboard.getNumber("DTL_throttleShape", m_throttleShape);
-
-    m_targetAngle = SmartDashboard.getNumber("DTL_targetAngle", m_targetAngle);
-    m_setPointDistance = SmartDashboard.getNumber("DTL_setPointDistance", m_setPointDistance);
-    m_angleThreshold = SmartDashboard.getNumber("DTL_angleThreshold", m_angleThreshold);
-    m_distThreshold = SmartDashboard.getNumber("DTL_distThreshold", m_distThreshold);
 
-    // load in Pid constants to controller
-    m_turnPid = new PIDController(m_turnPidKp, m_turnPidKi, m_turnPidKd);
-    m_throttlePid = new PIDController(m_throttlePidKp, m_throttlePidKi, m_throttlePidKd);
-
-    RobotContainer rc = RobotContainer.getInstance( );
-    rc.m_vision.m_tyfilter.reset( );
-    rc.m_vision.m_tvfilter.reset( );
-    rc.m_vision.syncStateFromDashboard( );
-  }
-
-  public void driveWithLimelightExecute( )
+  public Pose2d calculateTarget(VIConsts.VITargetLocations targetLocation)
```

Comment:
The driver button will cause the vision system to use the closest Target ID and return the "goal pose". Then the swerve system should accept the goal pose, generate a trajectory, and run the path (can use AutoDrivePath to accomplish this). Think about how this can be restructured to do this.